src/index.ts

```diff
@@ -138,6 +140,66 @@ export const createInstance = (config: Config): Instance => {
 
       return tx;
     },
+
+    buyNow: async (
+      signer: ethers.Signer,
+      params: BuyNowParams
+    ): Promise<ethers.ContractTransaction> => {
+      const zAuction = await getZAuctionContract(
+        signer,
+        config.zAuctionAddress
+      );
+
+      const nftContract = await getERC721Contract(
+        config.web3Provider,
+        config.tokenContract
+      );
+
+      const isApproved = await actions.isZAuctionApprovedNftTransfer(
+        await signer.getAddress(),
+        config.zAuctionAddress,
+        nftContract
+      );
+
+      if (!isApproved) throw Error("User is not approved for NFT Transfer");
+
+      const tx = await zAuction.buyNow(
+        params.auctionId,
+        params.amount,
+        params.tokenId,
+        params.startBlock,
+        params.expireBlock
+      );
+
+      return tx;
+    },
+
+    setBuyNow: async (
```

Comment:
Could we renamet his to `setBuyNowPrice` to be more clear that it's setting the price